Katello/katello
# Fixes #34635 - Duplicate upload of package

#### What are the changes introduced in this pull request?
Allow duplicate upload of package when a repo has the same package but synced with on demand.
#### Considerations taken when implementing this change?

#### What are the testing steps for this pull request?
1. Create a repo with content that's not been downloaded before and chose download policy as on demand. (Ex: https://fedorapeople.org/groups/katello/fakerepos/zoo/)
2. Download one of the packages from the repo locally. (ex: https://fedorapeople.org/groups/katello/fakerepos/zoo/cheetah-0.3-0.8.noarch.rpm)
3. Create a second repo and try to upload the cheetah package you downloaded.
4. It should succeed.
